eng/pipelines/templates/Scan_Code.yml

```diff
@@ -0,0 +1,38 @@
+steps:
+- task: PowerShell@1
+  displayName: "Print Environment Variables"
+  inputs:
+    scriptType: "inlineScript"
+    inlineScript: |
+      Get-ChildItem Env: | Sort-Object Name | Format-Table -Wrap -AutoSize
+
+- task: CredScan@2
+  inputs:
+    toolMajorVersion: "V2"
+
+- task: PowerShell@2
+  displayName: "Analyze CredScan artifact"
+  inputs:
+    targetType: 'inline'
+    script: |
+      try {
+        $credentialScannerMatches = Join-Path $(Pipeline.Workspace)\_sdt\logs\CredentialScanner CredentialScanner-matches.xml
+        $credScannerXml = [xml](Get-Content $credentialScannerMatches)
+
+        if($credScannerXml.DEFECTS -eq $null)
+        {
+          Write-Host "Unable to verify CodeAnalysisLogs artifact content, please manually verify it." 
+          exit 1          
+        }
+
+        if ($credScannerXml.DEFECTS.DEFECT -ne $null) {
+          Write-Host "CredentialScanner found following credential in this PR, please remove them from PR and use git squash/drop to remove commits with credential from git history then do force push:" 
```

Comment:
Are you sure squashing is adequate? My understanding is the commits still exist on the remote.
Anyone who obtained the commit id or github URL from a PR/etc can still access the original commit.

![image](https://user-images.githubusercontent.com/49205731/137248125-87c64896-bce8-4d92-83d6-36e095a25d6f.png)

https://github.com/NuGet/NuGet.Client/commit/5c96ea582e2cf75622ee771530799d1b255b6d2d

Reply:
You're right, yet can't think of better idea. I'm open for any other suggestions.
Even deleting actual github commit might not be enough,  not sure how often and what rate 3rd parties aggregate github.  For example [here's](https://www.gitmemory.com/NuGet/NuGet.Client) snapshot of nuget.

Reply:
> There's an article about it, mentioning tools I've not used myself: eg, `git filter-repo `. [Removing sensitive data from a repository](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository)

Most likely that tool is not enough in worst case scenario, someone or 3rd party bots already get the copy.

Reply:
Why not just:

> CredentialScanner found following credential in this PR, please revoke this credential immediately.